hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractParentQueue.java

```diff
@@ -490,6 +490,12 @@ public CSQueue createNewQueue(String childQueuePath, boolean isLeaf)
       String queueShortName = childQueuePath.substring(
           childQueuePath.lastIndexOf(".") + 1);
 
+      if (StringUtils.isEmpty(queueShortName)) {
+        throw new SchedulerDynamicEditException(
+                "Trying to create new queue=" + childQueuePath
+                  + ", which has empty leaf shortname.");
+      }
```

Comment:
Not sure about this. Going up the callchain: 

1. `AbstractParentQueue.createNewQueue` is only called from `ParentQueue.addDynamicChildQueue`, 
2. which (through another method) is called from `CapacitySchedulerQueueManager.createAutoQueue`, 
3. which is called from `CapacitySchedulerQueueManager.createQueue`, where my second comment should apply

There is one difference: upon determining whether we're using legacy auto creation or the flexible one we're using different methods, and the flexible auto queue creation (method: `CapacitySchedulerQueueManager.createQueue .createAutoQueue(queue)`) manipulates the QueuePath using `List<String> parentsToCreate = determineMissingParents(queue)`. If an empty string part can be added to the list, or can slip through I think it's probably there, which I think should be a better place for these checks.